### DISCUSSION TITLE
Placing icons back in dock after cask upgrades

### DISCUSSION THREAD

**Discussion by Danrancan** [Homebrew/discussions#4188] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.21-38-gc9496ae
ORIGIN: https://github.com/Homebrew/brew
HEAD: c9496ae50fefd506b1d75882d3283edc518bdf20
Last commit: 2 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9a3bb2f2da346d28fa6c431d835ef4c31f9f86c9
Core tap last commit: 2 weeks ago
Core tap branch: master
Core tap JSON: 09 Feb 16:21 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.4arRxMZ1fU/org.xquartz:0
HOMEBREW_EDITOR: nano
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 13.0.0 build 1300
Git: 2.32.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.7.3-x86_64
CLT: 13.2.0.0.1.1638488800
Xcode: 13.2.1
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  mtoc

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/bin/aft-mtp-cli
  /usr/local/bin/aft-mtp-mount
```

### Description of issue

When I run `brew upgrade --cask "myapp"` as in `brew upgrade google-chrome`, the cask upgrades, but "myapp" (google-chrome) is removed from the dock and I have to re-add it to the dock with every new version of each app. Is there a script or app that could constantly replace the dock icon with the new version of the cask after each upgrade? I have a ton of apps, and adding them all to the dock every time an upgrade happens is quite painstaking. Any advice or suggestions?

### Relevant casks

all of them

**Comment by cho-m** [comment]
This is a known issue https://github.com/Homebrew/homebrew-cask/issues/102721. You can check if any workarounds there work. Still pending on someone to contribute a PR to handle this properly.

**Reply by Danrancan** [reply]
Ahh, thank you! I guess I didn't search github thoroughly enough before posting. But thank you for the link and answer.
